src/java/org/apache/cassandra/metrics/ChunkCacheMetrics.java

```diff
@@ -7,79 +8,74 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
  */
 package org.apache.cassandra.metrics;
 
-import java.util.concurrent.TimeUnit;
 import javax.annotation.Nonnull;
 
-import com.codahale.metrics.Timer;
+import com.google.common.annotations.VisibleForTesting;
+
 import com.github.benmanes.caffeine.cache.stats.CacheStats;
 import com.github.benmanes.caffeine.cache.stats.StatsCounter;
 import org.apache.cassandra.cache.ChunkCache;
 
-import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
+import static org.apache.cassandra.config.CassandraRelevantProperties.USE_MICROMETER;
 
-/**
- * Metrics for {@code ICache}.
- */
-public class ChunkCacheMetrics extends CacheMetrics implements StatsCounter
+public interface ChunkCacheMetrics extends StatsCounter
 {
-    /** Latency of misses */
-    public final Timer missLatency;
-
-    /**
-     * Create metrics for the provided chunk cache.
-     *
-     * @param cache Chunk cache to measure metrics
-     */
-    public ChunkCacheMetrics(ChunkCache cache)
+    static ChunkCacheMetrics create(ChunkCache cache)
     {
-        super("ChunkCache", cache);
-        missLatency = Metrics.timer(factory.createMetricName("MissLatency"));
+        return USE_MICROMETER.getBoolean() ? new MicrometerChunkCacheMetrics(cache) : new CodahaleChunkCacheMetrics(cache);
     }
 
     @Override
-    public void recordHits(int count)
-    {
-        requests.mark(count);
-        hits.mark(count);
-    }
+    void recordHits(int count);
 
     @Override
-    public void recordMisses(int count)
-    {
-        requests.mark(count);
-        misses.mark(count);
-    }
+    void recordMisses(int count);
 
     @Override
-    public void recordLoadSuccess(long loadTime)
-    {
-        missLatency.update(loadTime, TimeUnit.NANOSECONDS);
-    }
+    void recordLoadSuccess(long loadTime);
 
     @Override
-    public void recordLoadFailure(long loadTime)
-    {
-    }
+    void recordLoadFailure(long loadTime);
 
     @Override
-    public void recordEviction()
-    {
-    }
+    void recordEviction();
+
```

Comment:
Another difference from the original, I removed methods `recordNotInCacheException()` and `notInCacheExceptions()` because the usage locations of those methods in DSE - `PrefetchingRebufferer.rebuffer()` do not exist in OSS currently.

If this is needed in cndb - maybe a future port will bring the code that needs it? - then perhaps it should be included here, however, I didn't want to bring over code that didn't have a clear usage.

Reply:
I don't think these are needed in CNDB.